test/test.getch.ts

```diff
@@ -381,6 +382,22 @@ describe('🥁 configuration options', () => {
     assert.deepStrictEqual(res.config.data, body);
   });
 
+  it('should not stringify the data if it is appended by a form', async () => {
+    const formData = new FormData();
+    formData.append('test', '123');
+    // I don't think matching formdata is supported in nock, so skipping: https://github.com/nock/nock/issues/887
+    const scope = nock(url).post('/').reply(200);
+    const res = await request({
+      url,
+      method: 'POST',
+      data: formData,
+    });
+    scope.done();
+    assert.deepStrictEqual(res.config.data, formData);
+    assert.ok(res.config.data instanceof FormData);
+    assert.deepEqual(res.config.body, undefined);
```

Comment:
@bcoe, I noticed last night that what is actually getting stringified (in the current way of doing things) [is the config body, not the data](https://github.com/googleapis/gaxios/blob/cc3dd894a9102692e389e7a7d529ac9898074ef4/src/gaxios.ts#L259). But in the issue, the author is describing the problem as stringifying the data. Therefore, I added an assertion to confirm that the body is not being set at all when the request contains a form. LMK if that looks right to you